src/vault/auth/approle.clj

```diff
@@ -24,6 +24,84 @@
     mount instead of the default. Passing `nil` will reset the client to the
     default.")
 
+  (upsert-role!
```

Comment:
Naming: I think a different verb here would be clearer as to the purpose - how about `configure-role!` instead?